### DISCUSSION TITLE
Include apps instead of exclude setting

### DISCUSSION THREAD

**Discussion by Nemipune** [wwmm/easyeffects#2635] [Q&A] [answered]
Hi ! Thanks for the amazing piece of software ! It's really simple and useful, I love it.

I was wondering if there's any way to approach the excluded app list as a whitelist instead of a blacklist ?

Like I only want to process audio from discord (pass it through a noise gate and a compressor) but I don't want those settings affecting any other software audio source, and manually excluding each one as they come would be painful.

Thanks a lot

**Comment by wwmm** [comment, marked as answer]
>I was wondering if there's any way to approach the excluded app list as a whitelist instead of a blacklist ?

No. The blocklist is only for blocking. But there is another way to get what you want. First disable `Process All Output Streams` in EasyEffects preferences Window. This way EasyEffects won't automatically redirect any audio stream anymore and you will have to manually enable effects for what you need.

The reason why this works is that wireplumber remembers the last device used by a player. So the next time you open discord it should be moved to EasyEffects virtual devices automatically. The downside of this approach is that sometimes wireplumber decides to move the stream somewhere different. But it does not do this kind of thing often.

**Reply by Nemipune** [reply]
All right it works, thanks a lot for the tip and explanations !
